src/extension.ts

```diff
@@ -29,43 +41,94 @@ const extract = async (options: {
     )
 
     const text = document.getText()
-    const { unbound, styledImported } = collectUnbound(text)
+    const unbound = collectUnbound(text)
     if (!unbound.length) {
       vscode.window.showWarningMessage(
-        '[SCE] Nothing to copy: There is no unbound components',
+        '[SCE] Nothing to extract: There are no unbound components',
       )
       return
     }
+
+    const exportIdentifier =
+      type == 'extractExportedToClipboard' || type == 'extractToSeparateFile'
+
     const declarations = generateDeclarations({
       unbound,
-      styledImported,
-      exportIdentifier: options.exportIdentifier,
-      importStyled: config.get('addImportStatement', true),
+      exportIdentifier,
     })
 
-    await vscode.env.clipboard.writeText(declarations)
-    vscode.window.showInformationMessage(
-      `[SCE] Copied to clipboard! (Found: ${unbound.length}) `,
-    )
+    if (type == 'extractToClipboard' || type == 'extractExportedToClipboard') {
+      let clipboardText = declarations
+      if (config.get('addImportStatement', true)) {
+        const styledImportInsertion = getStyledImportInsertion(
+          editor.document.getText(),
+        )
+        if (styledImportInsertion) {
+          clipboardText = styledImportInsertion.insertionText + declarations
+        }
+      }
+
+      await vscode.env.clipboard.writeText(clipboardText)
+
+      vscode.window.showInformationMessage(
+        `[SCE] Copied to clipboard! (Found: ${unbound.length}) `,
+      )
+    } else if (type == 'extractToSeparateFile') {
+      const styleFile = getCorrespondingStyleFile(
+        editor.document.uri.path,
+        config.get('separateFile.matchPattern', '*'),
+        config.get('separateFile.outputFile', 'styles.ts'),
+      )
+      if (!styleFile) {
+        vscode.window.showWarningMessage(
+          '[SCE] This file does not match the pattern in your configuration.',
+        )
+        return
+      }
+
+      await modifyImports(editor, styleFile, unbound)
+
+      const styleFileEditor = await openFileInEditor(styleFile)
+      await insertStyles(styleFileEditor, declarations)
+      await insertStyledImport(styleFileEditor)
+
+      await editor.document.save()
+      await styleFileEditor.document.save()
+    } else if (type == 'extractToSameFile') {
+      await insertStyles(editor, declarations)
+      await insertStyledImport(editor)
+
+      await editor.document.save()
+    }
   } catch (e) {
-    console.error('[SCE]', e)
-    vscode.window.showErrorMessage('[SCE] Failed to extract')
+    if (e instanceof Error && Object.getPrototypeOf(e).name === 'SyntaxError') {
+      vscode.window.showErrorMessage(
+        '[SCE] Failed to extract due to syntax error: ' + e.message,
+      )
+    } else {
+      console.error('[SCE]', e)
+      vscode.window.showErrorMessage('[SCE] Unexpected error while extracting')
+    }
```

Comment:
Bonus - if there was a parsing error (e.g. mismatched JSX tags), the process would previously fail with an error message. Added code to detect this specific case and provide a more helpful/actionable error message. This should increase the chances of the user fixing their own problems... rather than blaming the extension :-)